src/uproot/interpretation/strings.py

```diff
@@ -246,6 +327,32 @@ def final_array(
             library=library,
             branch=branch,
         )
+        if (
+            isinstance(library, uproot.interpretation.library.Awkward)
+            and self._special_case
+        ):
```

Comment:
```python
if any(not isinstance(x, StringArray) for x in basket_arrays.values()):
```

or the opposite of that for the StringArray case. Avoid extra attributes like `self._special_case`, especially mutable ones, especially if they don't have descriptive names. `:)`